### DISCUSSION TITLE
Analyzer should enable next analyzer for long running captures

### DISCUSSION THREAD

**Discussion by DevBruker** [gusmanb/logicanalyzer#48] [Ideas]
Hi,
use case: The signals of interest arrives in bursts of 50 - 120 ms of length. The bursts arrives at irregular times which can be from a few ms to several seconds apart. One analyzer can capture one burst maximum. This analyzer could then notify the next analyzer to be waiting for a trigger to happen. And when this second analyzer has finished it should enable a third one, and so on...  It is now necessary to set the sampling frequency down to 1 MHz to be able to capture the long bursts. But, 1 MHz is really too slow, and some short spikes are not sampled. Higher sampling frequency requires multiple analyzer where each analyzer captures a different part of the burst.
In this case it is ok to use some of the available capturing pins as "enabling" pins as a long sampling period is more important.

Regards

**Comment by gusmanb** [comment]
Hmmm, interesting idea.

I will think about it after the next release when I implement the replay functionality. 

I'm also thinking that it would be possible to use only one device, right now I have four buffers for the data with four chained DMA channels in ring mode, it may be possible to create a burst mode that captures up to four bursts, one per buffer, instead of four dma channels I can use a single one, then when a burst is captured the DMA is pointed to the next buffer and the trigger gets rearmed until all the burst have been captured. This is more feasible as it does not need changes to the triggers, the rearming would be responsability of the CPU and there I have more than enough room to do it.

As the burst in that case would be short it may be enough to have four burst of up to 32k samples (I assume this is for some kind of serial protocol where you will send/receive some commands so that would be 8 channel mode) without wasting samples waiting for the next data.

What do you think?

Cheers.

**Comment by DevBruker** [comment]
Yes, 8 channel mode is enough.                                                                                                                                                                                   
"it may be enough to have four burst of up to 32k samples"                                                                                                                                                       
One signal-burst in the use case lasts up to 120 ms. At 1 Mhz sampling 4 capturing-bursts lasts  about 130 ms which is enough to capture the complete signal-burst. At 10 Mhz 4 capturing-bursts would last about 13 ms which is not enough.                                                                                                                                                                      
                                                                                                                                                                                                                   
I would have thought that the main problem is to have enough time to transfer the data to the host before the next analyzer is filled up with data?                                                                                                                                                            
                                                                                                                                                                                                                   
And that it therefore is necessary to chain several analyzers like this:                                                                                                                                         
    1) Analyzer 1 fills up its capacity.                                                                                                                                                                           
    2) Analyzer 1 enables analyzer 2.                                                                                                                                                                              
    3) Analyzer 1 starts to transfer data to the host.                                                                                                                                                             
    4) Analyzer 2 is filled up and enables analyzer 3                                                                                                                                                              
    ...                                                                                                                                                                                                            
    x) Analyzer 1 finishes transferring data                                                                                                                                                                        
    ...                                                                                                                                                                                                            
    x) Analyzer 2 finishes transferring data                                                                                                                                                                        
    ...                                                                                                                                                                                                            
    x) Analyzer n enables analyzer 1.                                                                                                                                                                              
                                                                                                                                                                                                                   
The use case must decide the number of analyzers that allows the transferring of data to be completed in time before enabling the analyzer again.                                                                                                                                                      
                                                                                                                                                                                                                   This particular use case with 10 MHz sampling would require 10 analyzers (assuming one analyzer can hold 13 ms of data). After the 10th has finished sampling, about 130 ms after the trigger, there is a pause for about 5 seconds before a new round of up to 110 ms of data. During that 5 seconds all analyzers should have transferred data to the host and be ready for the next trigger.                                                                                                                                                                 
                                                                                                                                                                                                                   
The analyzer that is enabled should start capturing at once and not wait for a particular trigger condition. Except when the first one is enabled again...                                                                                                                                                                                                   
                                                                                                                                                                                                                   
Regards

**Comment by gusmanb** [comment]
Well, the thing is, does this make sense? 

I mean, the idea of this device is to have a cheap analyzer with good enough capabilities, if you're going to chain 10 of these analyzers, does it make sense to use this instead of buying a single one ten times the price of this one? Think that each analyzer is going to cost around 15$ including the level shifter, 10 analyzers would be 150$, for that price you can buy an analyzer that can handle millions of samples, so I don't see that function to fit on this project.

I have added the chain possibility only because if you have a couple of picos around you can chain them and have more channels, it is going to be still cheaper than anything else that is on the market but with its limitations. I allow up to five chained devices just in case that someone is so crazy to do it but the real idea is to chain a couple of them, no more, and get 48 channels that should be enough for most situations. Chaining 10 analyzers would be expensive, very voluminous and handling all those wires connected to a device would be a total nightmare, so for that case is better to just buy a better device.

In fact I have around something that can create that kind of analyzer and I have been thinking about developing it, Sipeed sells a very interesting board, the Tang Primer 20k, this board comes with a 20kLUT FPGA and 1Gbit of DDR3, with that a monster of analyzer can be developed supporting speeds of up to 500Mhz and millions of samples, and the best, the board preasembled costs 25€, so for 30-40€ you could get something that would crush a 500€ professional analyzer.

Cheers.
